WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/usecases/GetBackupDownloadStatusUseCaseTest.kt

```diff
@@ -169,6 +169,57 @@ class GetBackupDownloadStatusUseCaseTest : BaseUnitTest() {
                 assertThat(result).contains(progressStatus, completeStatus)
             }
 
+    @Test
+    fun `given max fetch retries exceeded, when backup status triggers, then return remote request failure`() =
+            coroutineScope.runBlockingTest {
+                whenever(activityLogStore.fetchBackupDownloadState(any()))
+                        .thenReturn(
+                            OnBackupDownloadStatusFetched(
+                                    BackupDownloadStatusError(GENERIC_ERROR),
+                                    FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+                            .thenReturn(
+                                    OnBackupDownloadStatusFetched(
+                                            BackupDownloadStatusError(GENERIC_ERROR),
+                                            FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+                            .thenReturn(
+                                    OnBackupDownloadStatusFetched(
+                                            BackupDownloadStatusError(GENERIC_ERROR),
+                                            FETCH_BACKUP_DOWNLOAD_STATE
+                            ))
+
+                val result = useCase.getBackupDownloadStatus(site, downloadId).toList()
+                advanceTimeBy(DELAY_MILLIS)
+
+                assertThat(result).contains(RemoteRequestFailure)
+            }
+
+    @Test
+    fun `given fetch error under retry count, when backup status triggers, then return progress`() =
+            coroutineScope.runBlockingTest {
+                whenever(activityLogStore.getBackupDownloadStatusForSite(site))
+                        .thenReturn(inProgressModel)
+                        .thenReturn(statusModel)
+                whenever(activityLogStore.fetchBackupDownloadState(any()))
+                        .thenReturn(
+                                OnBackupDownloadStatusFetched(
+                                        BackupDownloadStatusError(GENERIC_ERROR),
+                                        FETCH_BACKUP_DOWNLOAD_STATE
+                                ))
+                        .thenReturn(
+                                OnBackupDownloadStatusFetched(
+                                        BackupDownloadStatusError(GENERIC_ERROR),
+                                        FETCH_BACKUP_DOWNLOAD_STATE
+                                ))
+                        .thenReturn(OnBackupDownloadStatusFetched(FETCH_BACKUP_DOWNLOAD_STATE))
+
+                val result = useCase.getBackupDownloadStatus(site, downloadId).toList()
+                advanceTimeBy(DELAY_MILLIS)
+
+                assertThat(result).contains(progressStatus, completeStatus)
```

Comment:
💡 Idea: To make this test even more robust, we might want to consider using
assertThat(result).isEqualTo(listOf(progressStatus, completeStatus)) to make sure the usecase doesn't emit anything else. Wdyt?

Reply:
Made the change in 1983a4a. Better than using contains. Thanks